contracts/V1/AStateMachine.sol

```diff
@@ -60,6 +61,23 @@ abstract contract AStateMachine {
 
     function getZeroBalance() public pure virtual returns (Balance memory zeroBalance);
 
+    function processInboundMessage(Message calldata message) external _nonReentrant returns (bool) {
+        return _processInboundMessage(message);
+    }
+
+    function _processInboundMessage(Message calldata message) internal virtual returns (bool) {
+        if (message.messageType == MESSAGE_TYPE_JOIN) {
+            JoinChannel memory joinChannel = abi.decode(message.data, (JoinChannel));
```

Comment:
I get that naming things as inbount and outbound messages has elgance to it
but it seems a bit like a pointless abstrcation

the only inbound message is join and the only outbound is exit

also, if there are plans to expand to more kinds of messages (what kinds?)

then it would make more sens for a generla message to look like

{type: MessgeType, payload: Struct}



the feature we discussed, that a dispute is not neede in order to join, and only triggered if an inbound message was not digested, makes sense, bu that does not require the added abstraction